doc: add verifiers deployment doc (#152)
nervosnetwork/force-bridge: docs/verifiers-guide/deployment.md

# Deployment Guide For Verifiers

> In Force Bridge, assets of users are guarded by multi-signature of the committee. So every cross chain transaction

Reviewer: ```suggestion
In Force Bridge, the assets of users are guarded by the multi-signature of the committee. So every cross-chain transaction
```

will be signed by the committee to take effect on the counter-party chain.

> There will be two roles in the architecture, collector and verifier.

Reviewer: ```suggestion
There will be two roles in the architecture, **collector** and **verifier**.
```

> Collector will watch the Ethereum chain, compose mint transaction on CKB when it sees new confirmed lock transactions on

Reviewer: ```suggestion
The **collector** will watch the Ethereum chain, compose mint transactions on CKB when it sees newly confirmed lock transactions on
```

Ethereum, send the raw transaction to all verifiers to collect their signatures.

> Verifiers will watch both chains, verify the sign request sent by collector, provide their signature if it is a legal

Reviewer: ```suggestion
**Verifiers** will watch both chains, verify the sign request sent by the collector, provide their signature if it is a legal
```

> cross chain transaction.

Reviewer: ```suggestion
cross-chain transaction.
```

> When collector gets signatures more than multi-signature threshold, it will send the transaction to CKB Chain.

Reviewer: ```suggestion
When the collector gets signatures more than multi-signature threshold, it will send the transaction to CKB Chain.
```

> The process is similar to compose unlock transactions on Ethereum when it sees new confirmed burn transactions on CKB.

Reviewer: ```suggestion
The process is similar to compose unlock transactions on Ethereum when it sees newly confirmed burn transactions on CKB.
```

  rpc [options]
  verifier [options]
  config
  keystore
  help [command]       display help for command

# run CKB full node and indexer
# reference: <https://hub.docker.com/r/nervos/perkins-tent>
# you can change the port and mount directory as you want.
$ docker run -d -it -p 8117:9115  --name=ckb-testnet-indexer  -e "CKB_NETWORK=testnet" -v /path/to/ckb/data:/data nervos/perkins-tent:v0.43.0

# check if it works
echo '{
    "id": 2,
    "jsonrpc": "2.0",
    "method": "get_blockchain_info",
    "params": []
}' \
| tr -d '\n' \
| curl -H 'content-type: application/json' -d @- \
http://localhost:8117/rpc

echo '{
    "id": 2,
    "jsonrpc": "2.0",
    "method": "get_cells",
    "params": [
        {
            "script": {
                "code_hash": "0x9bd7e06f3ecf4be0f2fcd2188b23f1b9fcc88e5d4b65a8637b17723bbda3cce8",
                "hash_type": "type",
                "args": "0x8211f1b938a107cd53b6302cc752a6fc3965638d"
            },
            "script_type": "lock"
        },
        "asc",
        "0x64"
    ]
}' \
| tr -d '\n' \
| curl -H 'content-type: application/json' -d @- \
http://localhost:8117/indexer
```

## Generate Private Key and Associated Configs

> Generate your own private key. The private key is used to sign the cross chain transactions, keep it safe.

Reviewer: ```suggestion
Generate your own private key. The private key is used to sign the cross-chain transactions, keep them **SAFE**.
```

```bash
# generate the multiConfig
$ export PRIVKEY=0x1000000000000000000000000000000000000000000000000000000000000000
$ forcecli config generate -k $PRIVKEY
multiConfig:
{
  "ethAddress": "0x7B2419E0Ee0BD034F7Bf24874C12512AcAC6e21C",
  "ckbPubkeyHash": "0x42a34b11710e40b97180f8edae2760c0ab69bcf3",
  "ckbAddress": "ckt1qyqy9g6tz9csus9ewxq03mdwyasvp2mfhnesju9rhz"
}
# Send the multiConfig to the collector, it will be used in the deployment of on-chain contracts. 

# generate keystore
$ echo "{\"force-bridge\":\"${PRIVKEY}\"}" > keys.json
$ forcecli keystore encrypt -s ./keys.json -d ./keystore.json -p my-custom-password
# 1. Save the keystore.json file and remember your password.
# 2. Delete the keys.json file, close the terminal and clear the shell commands history to keep your private key safe.
```

## Run Verifier Service

> When we gets the multisig config from all verifiers, we will deploy the contracts, and then send verifiers a service

Reviewer 2: ```suggestion
When we gets the multisig config from all verifiers, we will deploy the contracts, and then run verifier  as a service
```

Author: it means we will send the verifiers the config, not run the service ourselves

| ckb.privateKey | the key of your private key in keystore. If you follow the previous steps, it will be `force-bridge` |

Run verifier service:

```bash
# start verifier service
$ forcecli verifier --config /path/to/your/config.json

# check if it works
$ echo '{ "id": 0, "jsonrpc": "2.0", "method": "status" }' \
| curl -H 'content-type: application/json' -d @- \
http://127.0.0.1:80/force-bridge/sign-server/api/v1
{
    "id": 0,
    "jsonrpc": "2.0",
    "result": {
        "addressConfig": {
            "ckbAddress": "ckt1qyqx9424gr3p237a36lt8veh50gaavc27jpqmhrdum",
            "ckbPubkeyHash": "0x62d55540e21547dd8ebeb3b337a3d1deb30af482",
            "ethAddress": "0x994B430359BEDCfeb73EB90Fb39416A7dE1F1640"
        },
        "latestChainStatus": {
            "ckb": {
                "latestCkbBlockHash": "0x95e1f7331c7a5a41ef24a8fbecc2ff10d2319fa678479d2345c8b8ebc04f9868",
                "latestCkbHeight": "1417"
            },
            "eth": {
                "latestEthBlockHash": "0xea3470c3ba5f26c4d049e9796940bb973258992d149e0f708774bac1b3182b7b",
                "latestEthHeight": "1180"
            }
        }
    }
}
```

> You can use your own process manager(systemd, pm2, supervisor, etc) to restart it when the process exits accidentally.

Reviewer: ```suggestion
You can use your own process manager(systemd, [pm2](https://pm2.keymetrics.io/), [supervisor](https://github.com/petruisfan/node-supervisor), etc) to restart it when the process exits accidentally.
```
